### PR TITLE
included shift_from_file

### PR DESCRIPTION
**Pull request by berkuva**

### CODE REVIEW

File: bedshift/bedshift.py [hunk 1]
```diff
@@ -295,6 +297,41 @@ def _shift(self, row, mean, stdev):
...
+            rows2shift = random.sample(list(range(len(intersect_regions))), num_shift)
+        except ValueError:
+            bedname = os.path.basename(self.bedfile_path)
+            shift_file_name = os.path.basename(fp)
+            print("The number of overlapping regions between "+
+                "{} and {} is {} but the shift ratio provided is trying to shift {} regions."\
+                .format(bedname, shift_file_name, len(intersect_regions), num_shift))
+            sys.exit(1)
+
+        shift_dict = {}
```
Comment (aaron-gu) [inline bedshift/bedshift.py:323]: At this point, can you modify the existing `shift` instead of writing a new version? So rather than modify `_shift`, we can add an argument that takes an existing set of `shift_rows` (instead of randomly sampling its own). That way, there will be less code changes, and we can keep `_shift` as is.

(What `_shift` is supposed to do is get one single region specified by `row`, then return a dictionary representing a new region. The old region is intended to be dropped.)

File: bedshift/bedshift.py [hunk 2]
```diff
@@ -392,12 +429,7 @@ def drop(self, droprate):
         return len(drop_rows)
 
     def _find_overlap(self, fp):
-        """
```
Comment (aaron-gu) [inline bedshift/bedshift.py:395]: Why is this deleted?

File: bedshift/bedshift.py [hunk 3]
```diff
@@ -652,6 +685,35 @@ def handle_yaml(self, bedshifter, yaml_fp):
                 num_changed += num_shifted
                 print("\t{} regions shifted.".format(num_shifted))
 
+            ##### shift_from_file #####
```
Comment (aaron-gu) [inline bedshift/bedshift.py:688]: At some point we should move this `blank from file` code to a separate file, and import it to use here.

File: bedshift/bedshift.py (hunk 2 above)
Comment (berkuva) [reply bedshift/bedshift.py:395]: I was just following that you did not have doc strings in the "private" functions that start with _, like _cut, but we can add this back in. I can include it back in my next pr.

File: bedshift/bedshift.py (hunk 3 above)
Comment (berkuva) [reply bedshift/bedshift.py:688]: Do you mean add_from_file, drop_from_file and shift_from_file, or the if/elif statements in handle_yaml?
